deluge/commonoptions.py

```diff
@@ -0,0 +1,143 @@
+        for l in text.splitlines():
+            wrapped_lines.extend(textwrap.wrap(l, width))
+        return wrapped_lines
+
+    def _format_action_invocation(self, action):
+        """
+        Combines the options with comma and displays the argument
+        value only once instead of after both options.
+        Instead of: -s <arg>, --long-opt <arg>
+        Show      : -s, --long-opt <arg>
+
+        """
+        if not action.option_strings:
+            metavar, = self._metavar_formatter(action, action.dest)(1)
+            return metavar
+        else:
+            parts = []
+            # if the Optional doesn't take a value, format is:
+            #    -s, --long
+            if action.nargs == 0:
+                parts.extend(action.option_strings)
+
+            # if the Optional takes a value, format is:
+            #    -s, --long ARGS
+            else:
+                default = action.dest.upper()
+                args_string = self._format_args(action, default)
+                opt = ", ".join(action.option_strings)
+                parts.append('%s %s' % (opt, args_string))
+            return ', '.join(parts)
+
+
+class HelpAction(argparse._HelpAction):
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if hasattr(parser, "subparser"):
+            subparser = getattr(parser, "subparser")
+            # If -h on a subparser is given, the subparser will exit after help message
+            subparser.parse_args()
+        parser.print_help()
+        parser.exit()
+
+
+class CommonOptionParser(argparse.ArgumentParser):
+
+    def __init__(self, *args, **kwargs):
+        if "formatter_class" not in kwargs:
+            kwargs["formatter_class"] = lambda prog: DelugeTextHelpFormatter(prog, max_help_position=33, width=90)
+        super(CommonOptionParser, self).__init__(*args, add_help=False, **kwargs)
+
+        self.common_setup = False
+        self.group = self.add_argument_group('Common Options')
+        self.group.add_argument('--version', action='version', version='%(prog)s ' + get_version(),
+                                help="Show program's version number and exit")
+        self.group.add_argument("-c", "--config", action="store", metavar='<config>',
+                                help="Set the config directory path")
+        self.group.add_argument("-l", "--logfile", action="store", metavar='<logfile>',
+                                help="Output to designated logfile instead of stdout")
+        self.group.add_argument("-L", "--loglevel", action="store", choices=deluge.log.levels, metavar='<level>',
+                                help="Set the log level: %s" % ", ".join(deluge.log.levels))
+        self.group.add_argument("-q", "--quiet", action="store_true", default=False,
+                                help="Sets the log level to 'none', this is the same as `-L none`")
+        self.group.add_argument("-r", "--logrotate", action="store", nargs="?", const="50M",
```

Comment:
Sorry I meant to add remove the `-r` as it's not an option that needs a shorthand. 

Perhaps split the max size to `--logrotatesize`?